src/utils/xhr.js

```diff
@@ -1,369 +0,0 @@
- * distributed under the License is distributed on an "AS-IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-import {assert} from './log';
-import {parseJson} from './json';
-import {parseUrl} from './url';
-import {utf8EncodeSync} from './bytes';
-
-/**
- * The "init" argument of the Fetch API. Currently, only "credentials: include"
- * is implemented.
- *
- * See https://developer.mozilla.org/en-US/docs/Web/API/GlobalFetch/fetch
- *
- * @typedef {{
- *   body: (!FormData|string|undefined),
- *   credentials: (string|undefined),
- *   headers: (!Object|undefined),
- *   method: (string|undefined),
- *   responseType: (string)
- * }}
- */
-export let FetchInitDef;
-
-/** @private @const {!Array<string>} */
-const allowedMethods_ = ['GET', 'POST'];
-
-/** @private @enum {number} Allowed fetch responses. */
-const allowedFetchTypes_ = {
-  document: 1,
-  text: 2,
-};
-
-/**
- * A class that polyfills Fetch API.
- */
-export class Xhr {
-  /**
-   * @param {!Window} win
-   */
-  constructor(win) {
-    /** @const {!Window} */
-    this.win = win;
-  }
-
-  /**
-   * We want to call `fetch_` unbound from any context since it could
-   * be either the native fetch or our polyfill.
-   *
-   * @param {string} input
-   * @param {!FetchInitDef} init
-   * @return {!Promise<!FetchResponse>|!Promise<!Response>}
-   * @private
-   */
-  fetch_(input, init) {
-    // TODO(avimehta): Should the requests go through when page is not visible?
-    assert(typeof input == 'string', 'Only URL supported: %s', input);
-    // In particular, Firefox does not tolerate `null` values for
-    // `credentials`.
-    const creds = init.credentials;
-    assert(
-      creds === undefined || creds == 'include' || creds == 'omit',
-      'Only credentials=include|omit support: %s',
-      creds
-    );
-    // Fallback to xhr polyfill since `fetch` api does not support
-    // responseType = 'document'. We do this so we don't have to do any parsing
-    // and document construction on the UI thread which would be expensive.
-    if (init.responseType == 'document') {
-      return fetchPolyfill(input, init);
-    }
-    return (this.win.fetch || fetchPolyfill).apply(null, arguments);
-  }
-
-  /**
-   * @param {string} input URL
-   * @param {?FetchInitDef} init Fetch options object.
-   * @return {!Promise<!FetchResponse>}
-   */
-  fetch(input, init) {
-    init = setupInit(init);
-    return this.fetch_(input, init)
-      .catch((reason) => {
-        /*
-         * If the domain is not valid for SwG we return 404 without
-         * CORS headers and the browser throws a CORS error.
-         * We include some helpful text in the message to point the
-         * publisher towards the real problem.
-         */
-        const targetOrigin = parseUrl(input).origin;
-        throw new Error(
```

Comment:
Preserve this behavior?

Reply:
Sure, I just added it here:
suggests devs look at Publisher Center setup when a fetch fails